src/lib/agent/PolykeyAgent.ts

```diff
@@ -515,7 +701,9 @@ class PolykeyAgent {
   public static async startAgent(daemon: boolean = false) {
```

Comment:
If this class is meant to represent the lifecycle of the `Agent`. The constructor of the `Agent` should already be starting the agent.

Also I prefer to have the filesystem operations outside of the construction. That is, I think the caller should be setting up the environment already. So the constructor just assumes the paths already exist. This minimizes of embedded side effects. Also allows asynchronous side effects to be used. I recommend whenever using `fs` to always use the async await versions. Also will be usable on nativescript.

At the same time, a deconstructor is necessary, but as we know it's not available: https://stackoverflow.com/questions/29333017/ecmascript-6-class-destructor

A proper cleanup will always be in the `finally` clause. So that should be in the entrypoint. This means all sideeffectful construction and destruction happens at the caller of the `Agent` and they do it in a `try { ... } finally { ... }` bracket.